src/__tests__/tigris.utility.spec.ts

```diff
@@ -122,6 +122,23 @@ describe("utility tests", () => {
 			expect(generated.getCollation().getCase()).toBe("ci");
 		});
 
+		const nerfingTestCases = [
+			["main/fork", "main_fork"],
+			["main-fork", "main-fork"],
+			["main?fork", "main?fork"],
+			["sTaging21", "sTaging21"],
+			["hotfix/jira-23$4", "hotfix_jira-23$4"],
```

Comment:
How would we handle the collision in the branch names? The user could have two different branches:
- hotfix/jira-23$4
- hotfix_jira-23$4

Reply:
The branch name is also in the http path. Would "$" need to be encoded?

Reply:
Collistions - We can potentially encode the names in path, but then every client has to be explicity encoding it. A better option would be to move this in the payload. this branch name is in payload for all the APIs except create/delete.


`$` doesn't need encoding as of now from my tests.
```
{'metadata': {}, 'collections': None, 'size': 0, 'branches': ['main', 'dependabotnpm_and_yarntypesnode-18.11.18', 'feature$123', 'feature_test', 'model%test', 'model-fork', 'rev\\perc', 'some space']}
```

Reply:
I think we should go with this for now.